src/qibolab/sweeper.py

```diff
@@ -98,6 +87,18 @@ def __post_init__(self):
             raise ValueError(
                 "Cannot create a sweeper without specifying pulses or channels."
             )
+        if self.range is not None and self.values is not None:
+            raise ValueError("'range' and 'values' are mutually exclusive.")
+        if self.range is None and self.values is None:
+            raise ValueError("Either 'range' or 'values' needs to be provided.")
+
+        return self
```

Comment:
The `model_post_init` is only one option, and it's the one to be used when:

> This is useful if you want to do some validation that requires the entire model to be initialized.
> https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_post_init

And most likely this is our case.

However, even `"after"` validators can do a very similar job, and, e.g., you can assign `Sweeper.values` here (if `None`), right after having checked that the two of them are not provided simultaneously.